cosmo/retrieval/find_new_cos_data.py

```diff
@@ -562,45 +516,82 @@ def tabulate_cache():
     to_copy_d = {}
     for key in list(missing_dict):
-        missing_files = missing_dict[key]
-        missing_in_cache = list(set(missing_files) & set(cache_a))
-        if len(missing_in_cache) == 0:
+        missing_files = missing_dict[key]  # missing files associated with PID
+        missing_in_cache = list(set(missing_files) & set(cache_roots))
+        # missing files that are in the cache
+        if len(missing_in_cache) == 0:  # if none of the missing files are
+            # in the cache, continue
             continue
         total_copied += len(missing_in_cache)
 
-        updated_missing = [x for x in missing_files if
-                           x not in missing_in_cache]
         updated_missing = list(set(missing_files) - set(missing_in_cache))
+        # update missing files list by removing files that were just
+        # determined to be in the cache
 
         if not updated_missing:
+            # if there is no difference in original missing list and new
+            # list based on searching for key = PID, this PID is misplaced
+            # and therefore is removed from the list of missing data sets
             missing_dict.pop(key,
                              "Something went terribly wrong, {0} isn't in "
                              "dictionary".format(
                                  key))
         else:
+            # if there are any remaining missing files associated with key =
+            # PID, set those in the missing dictionary
             missing_dict[key] = updated_missing
 
         # Create a generator where each element is an array with all 
-        # file types that match each missing dataset. Then concatenate all
-        # these
-        # individual arrays for ease of copying.
+        # file types that match each missing data set. Then concatenate all
+        # these individual arrays for ease of copying.
         # Joe said this makes sense, so it's ok, right?
         try:
-            to_copy = np.concatenate(tuple((cos_cache[np.where(cache_a == x)]
-                                            for x in missing_in_cache)))
+            # collect a tuple of all the files that are available to get from
+            # the cache for this key = PID
+            to_copy = np.concatenate(tuple((cos_cache
+                                            [np.where(cache_roots == x)] for
+                                            x in missing_in_cache)))
         except ValueError:
             print("RUH ROH!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+        # set tuple associated with each PID that is the list of all
+        # available files to copy to that PID
         to_copy_d[key] = to_copy
 
+    # missing PIDs left over
     end_missing = len(missing_dict.keys())
 
-    print("\tCopying {} total root(s) from cache, {} complete PID(s)".format(
-        total_copied, start_missing - end_missing))
+    print("\tPreparing to copy {} total root(s) from cache, {} complete PID("
```

Comment:
Could use formatted string literals here: 

-> `f'\t Preparing to copy {total_copied} total root(s) from cache, {start_missing - end_missing} complete PID(s)'`